### ISSUE TITLE
Floating point arguments broken in func references (win32, regression in 0.4.0)

### ISSUE THREAD

**Issue by hrydgard** [wasmerio/wasmer#415]
### Describe the bug

Calling functions with floating point parameters directly with ```instance.call("function_name",...);```  for example now works great, but calling it via ```Func.call``` doesn't, since 0.4.0. It worked in 0.3.0.

### Steps to reproduce

This can be reproduced easily in https://github.com/wasmerio/wasmer-rust-example/ :

Change Cargo.toml to use wasmer 0.4.0.

Then modify main.rs as follows:
```
instance.call("hello_wasm", &[wasmer_runtime::Value::F32(16.0)])?;
let f : wasmer_runtime::Func<(f32)> = instance.func("hello_wasm").unwrap();
f.call(16.0);
```
Also make the following edit to wasm-sample-app:
```
pub extern fn hello_wasm(f : f32) {
    // Call the function we just imported and pass in
    // the offset of our string and its length as parameters.
    unsafe {
      //print_str(HELLO.as_ptr(), HELLO.len());
      let s = format!("{}", f);
      print_str(s.as_ptr(), s.len());
    }
}
```
Compile both and see the output:
```
16
0.000000000000000000000000000000000000000000022
```
instead of the expected:
```
16
16
```

**Comment by hrydgard** [comment]
This is a pretty bad regression but it seems that there's currently no set of tests that could have caught it - your integration tests don't test hosting wasmer as a library in Rust and making this type of function call, as far as I can tell..

**Comment by bjfish** [comment]
Hi @hrydgard, thank you for this bug report. I am able to reproduce the on macOS well. I will look into to it.

**Comment by bjfish** [comment]
@hrydgard This issue should be fixed in master now that #435 is merged. Please try again and thanks for reporting this issue.

**Comment by hrydgard** [comment]
@bjfish  This is indeed now working as expected, thanks!
